crates/router/src/core/payments/routing.rs

```diff
@@ -281,31 +271,31 @@ async fn ensure_algorithm_cached_v1(
     #[cfg(not(feature = "business_profile_routing"))]
     let key = format!("dsl_{merchant_id}");
 
-    let present = ROUTING_CACHE
-        .present(&key)
-        .into_report()
-        .change_context(errors::RoutingError::DslCachePoisoned)
-        .attach_printable("Error checking presence of DSL")?;
-
-    let expired = ROUTING_CACHE
-        .expired(&key, timestamp)
-        .into_report()
-        .change_context(errors::RoutingError::DslCachePoisoned)
-        .attach_printable("Error checking expiry of DSL in cache")?;
+    let cached_algorithm = ROUTING_CACHE
+        .get_val::<Arc<CachedAlgorithm>>(key.as_str())
+        .await;
 
-    if !present || expired {
+    let algorithm = if let Some(algo) = cached_algorithm {
+        algo
+    } else {
```

Comment:
Return the routing algorithm from the `refresh_routing_cache_v1` function. Do the same for kgraph.